### DISCUSSION TITLE
Images in params.toml get sourced from doks node module

### DISCUSSION THREAD

**Discussion by cameronelliott** [thuliteio/doks#317] [Q&A] [answered]
I'm sorry if this should be obvious.
I went so far as to see how the showcase site Litestream.io handled my question, but the code didn't resolve my question.

Question:
When changing `config/_default/params.toml`, there are two variables: images, schemaLogo which
refer to `doks.png` and `logo-doks.png`, **so if I change them to my images, where do I put them???**

`doks.png` and `logo-doks.png` are found in `./node_modules/@hyas/doks/static`

Since they are under `node_modules`, obviously I shouldn't drop my images there.

**Likely resolution**
Sorry, I am a Hugo newbie.
I did a little digging/testing and the files in `./node_modules/@hyas/doks/static`
are copied to the root output folder of `public`.
And files in `static` also get copied into the `public`, so I should be able to put the images there.

**Question about the importance of the JSON-LD and Open Graph + Twitter Cards sections**
How important are the images in the JSON-LD and Open Graph + Twitter Cards sections ??
I don't have a good feel for how they are used. (nor the right sizes, etc)
(And the litestream.io site may have broken links in effect, because the specified png in his params.toml doesn't exist)

Thanks

**Comment by h-enk** [comment, marked as answer]
> When changing `config/_default/params.toml`, there are two variables: images, schemaLogo which
> refer to `doks.png` and `logo-doks.png`, **so if I change them to my images, where do I put them???**

In the `./static/` directory. The parent theme lives in `./node_modules/@hyas/doks/`, The child theme lives in `./`, following the structure of the parent theme. A child theme file overrides the corresponding parent theme file.

> And files in static also get copied into the public, ...

True.

>  .. so I should be able to put the images there.

Don't. The `public` directory is a temporary directory, and is deleted when starting the development server (`npm run start`), or building your site for production (`npm run build`).

> How important are the images in the JSON-LD and Open Graph + Twitter Cards sections ??
> I don't have a good feel for how they are used. (nor the right sizes, etc)

I'm aware that the current meta data setup can be way simpler.  I'm currently running some tests on my personal site — a simpler setup will be part of Doks v0.3 (coming soon).

> (And the litestream.io site may have broken links in effect, because the specified png in his params.toml doesn't exist)

Correct. Their [logo image](https://litestream.io/litestream.png) gives a 404, because .. it's not there (their [social media image](https://litestream.io/images/twitter-image.png) is in place).

**Reply by h-enk** [reply]
For background, this is the current logic for picking a social media image:

- If page parameter `images: []` is set — e.g. `images: ["image-1.png", "image-2.png"]`,  the first image is taken, so: `image-1.png`. Note, these images live in the corresponding page directory.
- If page parameter `images: []` is NOT set, the fallback image defined in `./config/_default/params.toml` is taken, e.g. `images = ["default-social-media-image.png"]`.
